flexmeasures/api/v3_0/tests/test_sensors_api.py

```diff
@@ -74,21 +76,19 @@ def test_fetch_sensors(
     if use_pagination:
         assert isinstance(response.json["data"][0], dict)
         assert is_valid_unit(response.json["data"][0]["unit"])
-        assert response.json["num-records"] == 3
-        assert response.json["filtered-records"] == 3
+        assert response.json["num-records"] == exp_num_results
+        assert response.json["filtered-records"] == exp_num_results
     else:
         assert isinstance(response.json, list)
         assert is_valid_unit(response.json[0]["unit"])
-        # NOTE: Even if the entire sensor name was passed to filter, due to the nature of the filter logic, the response will contain
-        # all sensors that have most of the keywords in the name. That's why we have 2 sensors in the response and I'm selecting the
-        # second one which has the name "some temperature sensor"
         if search_by == "unit":
             assert response.json[0]["unit"] == "°C"
             assert response.json[0]["name"] == "some temperature sensor"
+            assert len(response.json) == exp_num_results
         elif search_by == "filter":
-            assert response.json[1]["unit"] == "°C"
-            assert response.json[1]["name"] == "some temperature sensor"
-        assert len(response.json) == 2
+            assert response.json[0]["unit"] == "°C"
+            assert response.json[0]["name"] == "some temperature sensor"
+            assert len(response.json) == exp_num_results
```

Comment:
I see what you mean, its better for one of the other to be something different. 

how do you mean when you say the `exp_num_results` only needs to be tested once?

Reply:
>Maybe try a different unit, so the two test cases return something distinguishable?

by the way, this will mean I will need another conditional statement inside this block since the names of the assets are different for each unit.

some gas sensor = m³/h
some temperature sensor = °C

Cause of this I will need a conditional statement to check for the names on each test case


```
 if search_by == "unit":
            assert response.json[0]["unit"] == search_value
            assert len(response.json) == exp_num_results
            if search_value == "m³/h":
                assert response.json[0]["name"] == "some gas sensor"
            elif search_value == "°C":
                assert response.json[0]["name"] == "some temperature sensor"               
 ```

Reply:
> how do you mean when you say the exp_num_results only needs to be tested once?

Well, the line is fully parameterised, so you only need the line once.

> Cause of this I will need a conditional statement ...

You could of course also ass another test parameter `exp_sensor_name` :)

Reply:
ok, that's better actually